clients/redshift/staging.go

```diff
@@ -102,22 +114,35 @@ func (s *Store) loadTemporaryTable(tableData *optimization.TableData, newTableID
 			)
 
 			if err != nil {
-				return "", err
+				return "", nil, err
 			}
 
-			// TODO: Do something about result.NewLength
+			if result.NewLength > 0 {
+				_newLength, isOk := columnToNewLengthMap[col.Name()]
+				if !isOk || result.NewLength > _newLength {
+					// Update the new length if it's greater than the current one.
```

Comment:
If the column wasn't in `columnToNewLengthMap` then we could end up setting the length to something lower than it currently is. That could cause problems right?

Reply:
[NewLength](https://github.com/artie-labs/transfer/blob/ee1e09e38bbfcb667f4487972b9d07cd00ad39fb/clients/redshift/cast.go#L15) is only populated IFF:

1. expandStringPrecision is enabled (config)
2. Exceeded the current string precision
3. Less than Redshift's max character precision

Reply:
Anything else would result in NewLength = 0, which would not hit this block.

Reply:
Ah I see, I thought this was the only place we were checking that but now I see that this is just to handle the edge case where different precisions come through for different rows, to make sure we use the greater one. Carry on!